src/EndpointV2/Ruleset/RulesetStandardLibrary.php

```diff
@@ -0,0 +1,431 @@
+<?php
+
+namespace Aws\EndpointV2\Ruleset;
+
+use Aws\Arn\ArnParser;
+use Aws\Arn\Exception\InvalidArnException;
+use Aws\Exception\UnresolvedEndpointException;
+
+/**
+ * Provides functions and actions to be performed for endpoint evaluation.
+ * This is an internal only class and not subject to backwards compatibility guarantees.
+ *
+ * @internal
+ */
+class RulesetStandardLibrary
+{
+    const IPV4_RE = '/^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$/';
+    const IPV6_RE = '/([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|
+                    . ([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]
+                    . {1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:)
+                    . {1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|
+                    . [0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:
+                    . (:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|
+                    . 1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]
+                    . {1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]
+                    . |1{0,1}[0-9]){0,1}[0-9])/';
+    const TEMPLATE_ESCAPE_RE = '/{\{\s*(.*?)\s*\}\}/';
+    const TEMPLATE_SEARCH_RE = '/\{[a-zA-Z#]+\}/';
+    const TEMPLATE_PARSE_RE = '#\{((?>[^\{\}]+)|(?R))*\}#x';
+    const HOST_LABEL_RE = '/^(?!-)[a-zA-Z\d-]{1,63}(?<!-)$/';
+
+    private $partitions;
+
+    public function __construct($partitions)
+    {
+        $this->partitions = $partitions;
+    }
+
+    /**
+     * Determines if a value is set.
+     *
+     * @return boolean
+     */
+    public function is_set($value)
+    {
+        return isset($value);
+    }
+
+    /**
+     * Function implementation of logical operator `not`
+     *
+     * @return boolean
+     */
+    public function not($value)
+    {
+        return !$value;
+    }
+
+    /**
+     * Find an attribute within a value given a path string.
+     *
+     * @return mixed
+     */
+    public function getAttr($from, $path)
+    {
+        $parts = explode('.', $path);
+        foreach ($parts as $part) {
+            $sliceIdx = strpos($part, '[');
+            if ($sliceIdx !== false) {
+                if (substr($part, -1) !== ']') {
+                    return null;
+                }
+                # take the entire slice except for the last character (which is `]`)
```

Comment:
Why use # for comments here instead of //?

Reply:
idk, that was your comment 😛 .  Forgot to remove it though